modules/packages/Sort.chpl

```diff
@@ -47,24 +198,27 @@ private inline proc chpl_sort_cmp(a, b, param reverse=false, param eq=false) {
    :type doublecheck: `bool`
    :arg reverse: Sort in reverse numerical order
    :type reverse: `bool`
+   :arg comparator: Record that redefines the comparison mechanism with one of
+      the methods: ``comparator.key(a)`` or ``comparator.compare(a,b)``
 
  */
-proc BubbleSort(Data: [?Dom] ?eltType, doublecheck=false, param reverse=false) where Dom.rank == 1 {
+proc BubbleSort(Data: [?Dom] ?eltType, doublecheck=false, param reverse=false, comparator:?rec=emptyRecord) where Dom.rank == 1 && isRecord(rec) {
```

Comment:
I personally think that this `reverse` business is a waste of arguments.
Also doublecheck.

I can see keeping it for backwards compatibility, but I'd deprecate it.
For that reason I might prefer the actual sort algorithms be implemented with comparator and you just make other procs with the reverse argument that don't take in a comparator.
E.g. remove reverse and doublecheck from this one, and add
proc BubbleSort(Data, doublecheck=false, param reverse=false)
for backwards compatibility.

I'd probably criticize anybody who actually *calls* BubbleSort though. I'd personally advocate that we jettison it from the library completely.

But my comments will probably apply to the other sort variants.